users/models/custom_user.py

```diff
@@ -24,19 +26,24 @@ class CustomUser(AbstractUser):
     email = models.EmailField(blank=True, max_length=254,
                               verbose_name='email address', unique=True)
 
+    @atomic
     def save(self, **kwargs):
         """
         if img is too big we decrease img
         because the less image is the less memory it takes
         """
-        super(CustomUser, self).save(**kwargs)
-        if self.image:
-            img = Image.open(self.image.path)
-
-            if img.height > 300 or img.width > 300:
-                output_size = (300, 300)
-                img.thumbnail(output_size)
-                img.save(self.image.path)
+        try:
+            super(CustomUser, self).save(**kwargs)
+            if self.image:
+                img = Image.open(self.image.path)
+                if img.height > 300 or img.width > 300:
+                    output_size = (300, 300)
+                    img.thumbnail(output_size)
+                    img.save(self.image.path)
+        except (OSError, IOError):
+            self.image = None
+            super(CustomUser, self).save(update_fields=['image'])
+            raise PValidationError('Image can`t be saved')
```

Comment:
Please pay attention to the `atomic` decorator. An exception means rollback the transaction! I suppose it differs from result you'd may expected

Reply:
You mean i should use something like this?
```python
with transaction.atomic():
    super(CustomUser, self).save(**kwargs)
try:
    if self.image:
        ....

Reply:
No. In your sample transaction is meaningless